frame/support/src/traits/dispatch.rs

```diff
@@ -115,13 +116,90 @@ impl<OuterOrigin, L: EnsureOrigin<OuterOrigin>, R: EnsureOrigin<OuterOrigin>>
 	}
 }
 
+/// The "OR gate" implementation of `EnsureOrigin` from the different origins.
+///
+/// Origin check will pass if `L` or `R` origin check passes from the different origins. `L` is
+/// tested first.
+pub struct EnsureOneOfDifferentOrigins<L, R>(sp_std::marker::PhantomData<(L, R)>);
+
+impl<OuterOrigin1, OuterOrigin2, L: EnsureOrigin<OuterOrigin1>, R: EnsureOrigin<OuterOrigin2>>
+	EnsureOrigin<(OuterOrigin1, OuterOrigin2)> for EnsureOneOfDifferentOrigins<L, R>
```

Comment:
The implementation of `EnsureOrigin<(OneOrigin, AnotherOrigin)>` seems not useful, there is only one origin type defined for the runtime.
I don't see how there could be 2 different origin type.